core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -131,13 +133,29 @@ public BlockInfo getInfo(long blockId) throws IOException {
    * @return a stream which reads from the beginning of the block
    */
   public BlockInStream getInStream(long blockId, InStreamOptions options) throws IOException {
+    return getInStream(blockId, options, ImmutableSet.of());
+  }
+
+  /**
+   * Gets a stream to read the data of a block. This method is primarily responsible for
+   * determining the data source and type of data source. The latest BlockInfo will be fetched
+   * from the master to ensure the locations are up to date.
+   *
+   * @param blockId the id of the block to read
+   * @param options the options associated with the read request
+   * @param lostWorkers worker ids to exclude from the candidate list
+   * @return a stream which reads from the beginning of the block
+   */
+  public BlockInStream getInStream(long blockId, InStreamOptions options,
+      Set<WorkerNetAddress> lostWorkers) throws IOException {
```

Comment:
Lost workers can put incorporated into `InstreamOptions`? It should also be called `InstreamState` or something (unrelated to this PR)

Reply:
should we have both `InstreamOption` (immutable) and `InstreamState` (mutable)?

Reply:
but I am fine with leaving `lostWorkers` here as an arg if @bf8086 prefers this way